src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/RespondExtensionCallbackHandler.java

```diff
@@ -102,11 +102,12 @@ private CallbackResponse updateResponseDeadline(CallbackParams callbackParams) {
             data.put(RESPONSE_DEADLINE, newDeadline.atTime(MID_NIGHT));
         }
 
-        data.put("businessProcess",
-                 BusinessProcess.builder().activityId("ExtensionResponseHandling").status(READY).build());
+        List<String> errors = new ArrayList<>();
+        businessProcessService.updateBusinessProcess(data, "ExtensionResponseHandling", errors);
```

Comment:
hmm I don't like this updating errors, method is doing quite a lot. I think better if it returned a list of errors or an empty list

Reply:
Didn't want the method to return value and have side effect at the same time. But happy to change it.

Reply:
hmm you're right, it shouldn't return something and change something else. Trying to think how we can split this out

Reply:
should we maybe map businessProcess before sending to service?

Then one method to getProcessErrors(BusinessProcess) which returns errors or empty list. 
Second method which updates businessProccess with ID and ready state?

Then data.put() is still in the handler

Do you think the separate service is necessary or can it exist on the object?

Then we have some fluent code like businessProcess.getErrors(), businessProcess.updateToReadyStatus(activityId)

Reply:
decided to go for returning list of errors - it's the best option imho, as we don't copy-paste the same logic across all handlers